### PR TITLE
[KernelGen][Nvidia] Add randint operator with Triton kernel

### PR DESCRIPTION
**Pull request by XDYuanzhuLee**
## Summary
Adds a Triton kernel for `randint`. Returns a tensor filled with random integers generated uniformly between 0 (inclusive) and high (exclusive) using Philox random number generation.

## Testing
- Parametrized tests over SPECIAL_SHAPES and ALL_INT_DTYPES
- Validated output range: `0 <= output < high`, correct shape and dtype
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_randint.py --level core` (NVIDIA H20, dtype=torch.float32)

| Configuration | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| [1073741824] | 4.540 | 3.882 | 1.170 |
| [64, 64] | 0.007 | 0.008 | 0.875 |
| [4096, 4096] | 0.080 | 0.069 | 1.164 |
| [64, 512, 512] | 0.081 | 0.069 | 1.179 |
| [1024, 1024, 1024] | 4.540 | 3.882 | 1.170 |
| **Arithmetic Mean** | — | — | **1.112** |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (5 cases, --level core) | 1.112 | Primary |
| Tianshu | PASS | PASS (18 cases) | 1063.911 | — |
| Muxi | PASS | PASS (18 cases) | 0.955 | — |
| Ascend | FAIL | N/A | — | metadata_path issue |
| Hygon | PASS | PASS (18 cases) | 1.022 | — |

## Files Changed
- `src/flag_gems/ops/randint.py`: Triton kernel implementation
- `tests/test_randint.py`: Accuracy test
- `benchmark/test_randint.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Distribution, stage: alpha 5.1)

### CODE REVIEW

File: src/flag_gems/ops/__init__.py [hunk 1]
```diff
@@ -738,6 +739,7 @@
     "rand_like",
     "randn",
     "randn_like",
+    "randint",
```
Comment (tengqm) [inline src/flag_gems/ops/__init__.py]: can we keep this list sorted?

File: benchmark/test_randint.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import pytest
+import torch
+
+from . import base
+
+
+def _input_fn(shape, dtype, device):
```
Comment (tengqm) [inline benchmark/test_randint.py]: where do you get shape, dtype and device?

File: src/flag_gems/ops/__init__.py (hunk 1 above)
Comment (XDYuanzhuLee) [reply src/flag_gems/ops/__init__.py]: Done. Moved `randint` before `randn` in `__all__` to maintain alphabetical order.

File: benchmark/test_randint.py (hunk 2 above)
Comment (XDYuanzhuLee) [reply benchmark/test_randint.py]: Good catch — replaced with the worktree version: `RandintBenchmark(GenericBenchmarkExcluse1D)` with proper `randint_input_fn(shape, dtype, device)` that yields `(high, shape)`. Also switched dtypes to `consts.INT_DTYPES`.

**Review by w1120029931-bit — APPROVED** [review]
